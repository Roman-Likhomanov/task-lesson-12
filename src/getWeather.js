export async function getWeather(cityName) {
  const apiKey = "feeab8e8e1e7b8d68fab64472693b6f7";
  const response2 = await fetch(
    `https://api.openweathermap.org/data/2.5/weather?q=${cityName}&appid=${apiKey}`
  );

  const data = await response2.json();

  if (data.cod === "404") {
    throw new Error(alert("Пожалуйста, введите город корректно!"));
  }

  return data;
}
