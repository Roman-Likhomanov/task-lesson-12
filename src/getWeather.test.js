import { getWeather } from "./getWeather.js";

describe("getWeather", () => {
  global.fetch = jest.fn(() =>
    Promise.resolve({
      json: () => Promise.resolve({ cod: "404" }),
    })
  );

  it("throw error", async () => {
    window.alert = jest.fn();
    await expect(getWeather("ergrg")).rejects.toThrowErrorMatchingSnapshot();
    expect(window.alert).toHaveBeenCalledWith(
      "Пожалуйста, введите город корректно!"
    );
  });
});
