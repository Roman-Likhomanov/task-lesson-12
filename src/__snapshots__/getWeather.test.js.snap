// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getWeather throw error 1`] = `""`;
